src/propTypes.ts

```diff
@@ -146,14 +149,16 @@ function createObservableTypeCheckerCreator(allowNativeType, mobxType) {
     })
 }
 
-function createObservableArrayOfTypeChecker(allowNativeType, typeChecker) {
+function createObservableArrayOfTypeChecker(
+    allowNativeType: boolean,
+    typeChecker: React.Validator<any>
+) {
     return createChainableTypeChecker(function(
-        props,
-        propName,
-        componentName,
-        location,
-        propFullName,
-        ...rest
```

Comment:
I removed the `...rest` spread here since it wasn't being used and didn't match the shape of React.Validator.  Hopefully that is okay?